packages/slice-machine/server/src/api/io/Slice.ts

```diff
@@ -11,8 +12,19 @@ export function writeSlice(path: string, slice: SliceSM) {
   Files.write(path, Slices.fromSM(slice));
 }
 
-export function renameSlice(path: string, newSliceName: string) {
-  const slice = readSlice(path);
+export function writeSliceTypes(path: string, slice: SliceSM) {
+  const content = generateTypes({ sharedSliceModels: [Slices.fromSM(slice)] });
+
+  Files.write(path, content);
+}
+
+export function renameSlice(
+  modelPath: string,
+  typesPath: string,
+  newSliceName: string
+) {
+  const slice = readSlice(modelPath);
   slice.name = newSliceName;
-  writeSlice(path, slice);
+  writeSlice(modelPath, slice);
+  writeSliceTypes(typesPath, slice);
```

Comment:
same question as the rename custom types, why do we need to rebuild the types if we rename them ?

Reply:
TSDocs on the generated type make use of the Slice Display name here too ☺️